src/syntax/FStar.Syntax.Util.fs

```diff
@@ -997,24 +997,28 @@ let let_rec_arity (lb:letbinding) : int * option<(list<bool>)> =
        Common.tabulate n_univs (fun _ -> false)
        @ (bs |> List.map (fun b -> U.set_mem b.binder_bv d_bvs)))
 
-let abs_formals t =
+let abs_formals_maybe_unascribe_body maybe_unascribe t =
     let subst_lcomp_opt s l = match l with
         | Some rc ->
           Some ({rc with residual_typ=FStar.Util.map_opt rc.residual_typ (Subst.subst s)})
         | _ -> l
     in
     let rec aux t abs_body_lcomp =
-        match (unascribe <| Subst.compress t).n with
+        match (unmeta_safe t).n with
```

Comment:
This is a small but subtle change. Unascribe would rip off meta nodes, even if they were relevant. Switched it to use unmeta_safe, retaining both ascriptions and semantically relevant meta nodes when unpacking an abstraction